packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FWFNavigationDelegateHostApi.m

```diff
@@ -26,15 +86,15 @@ - (FWFNavigationDelegate *)navigationDelegateForIdentifier:(NSNumber *)instanceI
 }
 
 - (void)createWithIdentifier:(nonnull NSNumber *)instanceId
-                       error:(FlutterError *_Nullable *_Nonnull)error {
-  FWFNavigationDelegate *navigationDelegate = [[FWFNavigationDelegate alloc] init];
+    didFinishNavigationIdentifier:(nullable NSNumber *)didFinishNavigationInstanceId
+                            error:(FlutterError *_Nullable __autoreleasing *_Nonnull)error {
+  FWFNavigationDelegate *navigationDelegate =
+      [[FWFNavigationDelegate alloc] initWithBinaryMessenger:self.binaryMessenger
+                                             instanceManager:self.instanceManager];
   [self.instanceManager addInstance:navigationDelegate withIdentifier:instanceId.longValue];
-}
-
-- (void)setDidFinishNavigationForDelegateWithIdentifier:(nonnull NSNumber *)instanceId
-                                     functionIdentifier:(nullable NSNumber *)functionInstanceId
-                                                  error:(FlutterError *_Nullable __autoreleasing
-                                                             *_Nonnull)error {
-  // TODO(bparrishMines): Implement when callback method design is finalized.
+  if (didFinishNavigationInstanceId) {
+    [self.instanceManager addInstance:navigationDelegate.didFinishNavigation
+                       withIdentifier:didFinishNavigationInstanceId.longValue];
```

Comment:
Having to do object tracking for every callback definitely makes the subclass-and-override approach to delegates more appealing. But if finalizer behavior makes that impractical, I guess we're stuck with this.

Reply:
I definitely agree. So I looked for another solution and the comment you made about `copyWith` in the doc gave me a new idea that wouldn't require tracking any callback methods.

It keeps the same concept of all classes being stored as weak references in the Dart `InstanceManager` and strong references in the `InstanceManager` of the native language.

But instead of storing the Dart functions as strong references in the Dart `InstanceManager`, a copy of the Dart class is stored as a strong reference. The copied instance would keep the functions passed in the original instance reachable by the Flutter api.

And on the native side, the **instance** is also stored as a weak reference rather than each function. 

This pattern would still allow
1. Dart weak reference to release the native strong reference.
2. The native weak reference would release the copy of the the Dart strong reference.

Reply:
The `copy` method should also be made as a library private method. I'm still on the fence of annotating it with `@protected` or `@internal`.